functions/go/apply-setters/walk.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// visitor is implemented by structs which need to walk the configuration.
+// visitor is provided to accept to walk configuration
+type visitor interface {
+	// visitScalar is called for each scalar field value on a resource
+	// node is the scalar field value
+	visitScalar(node *yaml.RNode) error
+
+	// visitMapping is called for each Mapping field value on a resource
+	// node is the mapping field value
+	visitMapping(node *yaml.RNode) error
+}
+
+// accept invokes the appropriate function on v for each field in object
+func accept(v visitor, object *yaml.RNode) error {
+	return acceptImpl(v, object)
+}
+
+// acceptImpl implements accept using recursion
+func acceptImpl(v visitor, object *yaml.RNode) error {
+	switch object.YNode().Kind {
+	case yaml.DocumentNode:
+		// Traverse the child of the document
+		return accept(v, yaml.NewRNode(object.YNode()))
```

Comment:
OOC: why do you create a new RNode?

Reply:
Inherited some old code here. Thanks for pointing.